### PR TITLE
Telemetryv3

### PR DESCRIPTION
**Pull request by anujtech46**
1- Added telemetry for portal backend
2- Added telemetry for the announcement

### CODE REVIEW

**Comment by harishgilimi** [comment]
@anujtech46  resolve merge conflits

**Comment by anujtech46** [comment]
@harishgilimi, I did the required changes, Let me know if any issue.

File: src/app/helpers/permissionsHelper.js [hunk 1]
```diff
@@ -54,11 +55,18 @@ let PERMISSIONS_HELPER = {
       }
     }
     request(options, function (error, response, body) {
-      if (!error && body) {
-        body = JSON.parse(body)
-        if (body.responseCode === 'OK') {
-          module.exports.setRoleUrls(body.result)
-        }
+      body = JSON.parse(body)
```
Comment (harishgilimi) [inline src/app/helpers/permissionsHelper.js]: Move this up before request

File: src/app/helpers/permissionsHelper.js [hunk 2]
```diff
@@ -96,9 +104,19 @@ let PERMISSIONS_HELPER = {
     request(options, function (error, response, body) {
       reqObj.session.roles = []
       reqObj.session.orgs = []
+      body = JSON.parse(body)
```
Comment (harishgilimi) [inline src/app/helpers/permissionsHelper.js]: Move this up before request

File: src/app/helpers/publicServiceHelper.js [hunk 3]
```diff
@@ -23,9 +25,20 @@ module.exports = {
...
     }
     request(options, function (error, response, body) {
+      // Add telemetry log event
+      const telemetryData = {reqObj: req,
+        options: options,
+        statusCode: response.statusCode,
+        resp: body,
+        uri: 'org/v1/search',
+        userId: 'Public'}
+      telemetryHelper.logAPICallEvent(telemetryData)
```
Comment (harishgilimi) [inline src/app/helpers/publicServiceHelper.js]: Move this up before request

**Comment by anujtech46** [comment]
@harishgilimi, I did the required changes, Let me know if any issue.

**Comment by anujtech46** [comment]
@harishgilimi, Added roll-up data.

**Pull request merged**